components/cards/EventCard.vue

```diff
@@ -0,0 +1,130 @@
+<template>
+  <nuxt-link
+    :to="to"
+    class="card-link"
+  >
+    <article class="event-card">
+      <div class="event-card__content">
+        <header>
+          <span class="event-card__subtitle">
+            {{ type }}
+          </span>
+          <h3 class="event-card__title">
+            {{ title }}
+          </h3>
+        </header>
+
+        <footer>
+          <div class="event-card__info-detail">
+            <Map20 class="event-card__icon-map" />
+            <span class="event-card__place">{{ place }}</span>
+          </div>
+          <div class="event-card__date-and-arrow">
+            <div class="event-card__info-detail">
+              <Calendar20 class="event-card__icon-calendar" />
+              <span class="event-card__date">{{ date }}</span>
+            </div>
+            <ArrowRight20 class="event-card__icon-arrow-right" />
+          </div>
+        </footer>
+      </div>
+
+      <div
+        class="event-card__media"
+        :style="`background-image: ${decorate(image)};`"
+      />
+    </article>
+  </nuxt-link>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Prop } from 'vue-property-decorator'
+
+@Component
+export default class extends Vue {
+  @Prop(String) type
+  @Prop(String) title
+  @Prop(String) image
+  @Prop(String) place
+  @Prop(String) date
+  @Prop(String) to
+
+  decorate (image) {
+    const bgEffects = [
+      `url(${image})`
+    ]
+    return bgEffects.join(', ')
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+@mixin config-icons($prefix, $colors...) {
```

Comment:
It's cool to use SCSS mixins but in this case you don't need them and you ended up generating more unnecessary CSS.

If you know that
```
     position: relative;
      top: .3em;
      width: 1.25rem;
      height: 1.25rem;
```
are shared thought all the icons, you should apply to all of them the class `event-card__icon` with the shared properties and then add a BEM's modifier on this class with the prefix `--`. Check more about BEM's modifier [here](http://getbem.com/naming/).


My proposed solution:

**HTML**
```
<Map20 class="event-card__icon" />
<Calendar20 class="event-card__icon" />
<ArrowRight20 class="event-card__icon event-card__icon--purple" />

```
**CSS**
```
.event-card__icon {
      fill: currentColor;
      position: relative;
      top: .3em;
      width: 1.25rem;
      height: 1.25rem;
}
.event-card__icon--purple {
      color: var(--purple-60)
}
```

Notice that I'm using `currentColor`. This will allow you to get the parent color for your SVG.